server/src/main/java/org/elasticsearch/cluster/metadata/DesiredNode.java

```diff
@@ -226,4 +251,88 @@ public String toString() {
             + ']';
     }
 
+    public record Processors(float min, float max) implements Writeable, ToXContentObject {
+
+        private static final Version MIN_MAX_PROCESSORS_VERSION = Version.V_8_3_0;
+
+        private static final ParseField MIN_FIELD = new ParseField("min");
+        private static final ParseField MAX_FIELD = new ParseField("max");
+
+        public static final ConstructingObjectParser<Processors, String> PROCESSORS_PARSER = new ConstructingObjectParser<>(
+            "processors",
+            false,
+            (args, name) -> new Processors((float) args[0], (float) args[1])
+        );
+
+        static {
+            PROCESSORS_PARSER.declareFloat(ConstructingObjectParser.constructorArg(), MIN_FIELD);
+            PROCESSORS_PARSER.declareFloat(ConstructingObjectParser.constructorArg(), MAX_FIELD);
```

Comment:
Should this be optional? I think k8s might only have requests set, leading to no max cpu. In that case we would still like to know the minimum value.